File: frontend/src/services/user.service.js
import api from './api';

class UserService {
  getUserProfile() {
    return api.get('/users/me');
  }

  updateUserProfile(id, userData) {
    return api.put(`/users/${id}`, userData);
  }

  getUserById(id) {
    return api.get(`/users/${id}`);
  }
}

export default new UserService();

File: backend/src/main/java/com/chami/backend/services/UserService.java
package com.chami.backend.services;

import com.chami.backend.dtos.UserDTO;

public interface UserService {
    UserDTO getUserProfile(Long userId);
    UserDTO updateUserProfile(Long userId, UserDTO userDTO);
    void deleteUser(Long userId);
    UserDTO getCurrentUser(String username);
}
